httpclient/config.go

```diff
@@ -0,0 +1,422 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"errors"
+	"github.com/acronis/go-appkit/config"
+	"github.com/acronis/go-appkit/retry"
+	"github.com/cenkalti/backoff/v4"
+	"time"
+)
+
+// RetryPolicy represents a retry policy strategy.
+type RetryPolicy string
+
+const (
+	// DefaultReqType is a default request type.
+	DefaultReqType = "go-appkit"
+
+	// DefaultClientWaitTimeout is a default timeout for a client to wait for a request.
+	DefaultClientWaitTimeout = 10 * time.Second
+
+	// RetryPolicyExponential is a policy for exponential retries.
+	RetryPolicyExponential RetryPolicy = "exponential"
+
+	// RetryPolicyConstant is a policy for constant retries.
+	RetryPolicyConstant RetryPolicy = "constant"
+
+	// configuration properties
+	cfgKeyRetriesEnabled                          = "retries.enabled"
+	cfgKeyRetriesMax                              = "retries.maxAttempts"
+	cfgKeyRetriesPolicy                           = "retries.policy"
+	cfgKeyRetriesPolicyExponentialInitialInterval = "retries.exponentialBackoff.initialInterval"
+	cfgKeyRetriesPolicyExponentialMultiplier      = "retries.exponentialBackoff.multiplier"
+	cfgKeyRetriesPolicyConstantInternal           = "retries.constantBackoff.interval"
+	cfgKeyRateLimitsEnabled                       = "rateLimits.enabled"
+	cfgKeyRateLimitsLimit                         = "rateLimits.limit"
+	cfgKeyRateLimitsBurst                         = "rateLimits.burst"
+	cfgKeyRateLimitsWaitTimeout                   = "rateLimits.waitTimeout"
+	cfgKeyLogEnabled                              = "log.enabled"
+	cfgKeyLogMode                                 = "log.mode"
+	cfgKeyLogSlowRequestThreshold                 = "log.slowRequestThreshold"
+	cfgKeyMetricsEnabled                          = "metrics.enabled"
+	cfgKeyTimeout                                 = "timeout"
+)
+
+var _ config.Config = (*Config)(nil)
+var _ config.KeyPrefixProvider = (*Config)(nil)
+
+// RateLimitConfig represents configuration options for HTTP client rate limits.
+type RateLimitConfig struct {
```

Comment:
`RateLimitConfig` -> `RateLimitsConfig`?
This would ensure consistency with `RetriesConfig` and `MetricsConfig`.